### PR TITLE
[Manual backport 1.x] Gets rid of `github-checks-reporter` (#3126)

### PR DESCRIPTION
**Pull request by zhongnansu**
Signed-off-by: Miki <miki@amazon.com>
(cherry picked from commit 5116f1619b7deb76fbefd1c91b346ec2aef2b584)

Coming from https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3126

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: tasks/config/run.js [hunk 1]
```diff
@@ -40,107 +40,82 @@ const OPENSEARCH_DASHBOARDS_INSTALL_DIR =
...
-    }),
+    },
 
-    sasslint: scriptWithGithubChecks({
-      title: 'sasslint',
+    stylelint: {
       cmd: NODE,
-      args: ['scripts/sasslint'],
-    }),
+      args: ['scripts/stylelint'],
```
Comment (AMoo-Miki) [inline tasks/config/run.js:54]: this should remain as `sasslint` for 1.x.

File: yarn.lock [hunk 2]
```diff
@@ -9125,6 +9007,21 @@ execa@^4.0.0, execa@^4.0.2:
     signal-exit "^3.0.2"
     strip-final-newline "^2.0.0"
 
+execa@^4.0.2:
```
Comment (AMoo-Miki) [inline yarn.lock:9010]: Can you please try removing both of these `execa@^4.0.0` and `execa@^4.0.2`, and do a bootstrap to fix this section.

**Review by AMoo-Miki — CHANGES_REQUESTED** [review]

File: tasks/config/run.js [hunk 3]
```diff
@@ -40,107 +40,82 @@ const OPENSEARCH_DASHBOARDS_INSTALL_DIR =
...
+    },
```
Comment (AMoo-Miki) [inline tasks/config/run.js:55]: This should remain as `sasslint`; simply replace `stylelint` with `sasslint`:
```suggestion
    sasslint: {
      cmd: NODE,
      args: ['scripts/sasslint'],
    },
```

**Review by AMoo-Miki — CHANGES_REQUESTED** [review]

File: tasks/config/run.js (hunk 3 above)
Comment (zhongnansu) [reply tasks/config/run.js]: thx, made the update

**Review by AMoo-Miki — APPROVED** [review]

**Review by kristenTian — APPROVED** [review]

**Pull request merged**